app/Models/Concerns/HasFilters.php

```diff
@@ -421,7 +424,53 @@ protected function filterRaces(Builder $query, string $value = null): void
             ;
         }
     }
+    /**
+     * Filter on characters on multiple locations
+     * @param Builder $query
+     * @param string|null $value
+     * @return void
+     */
+    protected function filterLocations(Builder $query, string $value = null): void
+    {
+        $key = 'location_id';
+        if ($this->filterOption('exclude')) {
```

Comment:
A lot of this is duplicate code, is it not possible to call the fallback function if it's not `children`?